bsip-0044.md

```diff
@@ -86,41 +92,46 @@ Upon presentation of a request from the `depositor` for return of `escrow assets
 
 The `timeout threshold` of the contract is defined by `depositor` within the HTLC at creation. It can be any time in the future and should allow enough time for `recipient` to review the HTLC and provide the `preimage`. Further, it should not be set too far into the future to mitigate against an unresponsive `recipient` impacting `depositor`, as their `escrow assets` will be locked until `timeout threshold` expiry. The accuracy is based on when the `condition evaluator` runs, and should be considered accurate ± 15 seconds.
 
+**Note:** we recommend the Committee set the maximum value for `timeout threshold` to limit the amount of time a contract may consume memory of validation nodes.
+
 ### **Early Termination of an Escrow Contract**
 
-To protect the `recipient`, early termination of an HTLC is not allowed by any party. Placing a `timeout threshold` far into the future is valid. User protection from locking up funds for an extremely long period could be provided by the UI used to create the HTLC.
+To protect the `recipient`, early termination of an HTLC is not allowed by any party. Placing a `timeout threshold` far into the future is valid, up to the maximum defined by the Committee. User protection from locking up funds for an extremely long period could be provided by the UI used to create the HTLC.
 
 ### **Automatic Transfers Upon Expiry**
 
-Upon expiry of the `timeout threshold`, the `escrow assets` held within the HTLC will **not** automatically be returned to `depositor`. A separate transaction must be initiated by the `depositor` to request `escrow assets` be transferred to `depositor`. 
+Upon expiry of the `timeout threshold`, the `escrow assets` held within the HTLC will be queued for return to `depositor`. No action is required by the `depositor` to receive their "locked" funds back from the contract after expiry. 
```

Comment:
Please clarify: an expired HTLC that is "queued for return" can no longer be claimed by recipient, even if it hasn't been returned to depositor yet.

Reply:
Consider this insertion after the first sentence:
"From this time, the HTLC will no longer evaluate the `hash lock`, preventing `recipient` from receiving the `escrow assets`. "